### PR TITLE
[algebras/quaternion.py] Docstrings added

### PR DESCRIPTION
**Pull request by GaurangTandon**
#### References to other Issues or PRs

#15370

#### Brief description of what is fixed or changed

More docstrings added. 

#### Other comments

I intend to add more documentation in patches (that way it is also easy to review and maintain). So, this is the third patch after my previous PR.

#### Release Notes

* algebras 
  * Improve docstring for `from_axis_angle`, `from_rotation_matrix`, `add`, `mult`, and `pow` in `quaternion.py`

### CODE REVIEW

File: sympy/algebras/quaternion.py [hunk 1]
```diff
@@ -108,8 +122,18 @@ def from_axis_angle(cls, vector, angle):
...
-        only if the matrix is special orthogonal (orthogonal and det(M) = 1).
+        """Returns the equivalent quaternion of a matrix.
+
+        Parameters
+        ==========
+        M : Matrix
+            The matrix from which an equivalent quaternion has to be obtained.
+
+        Returns
+        ==========
```
Comment (oscarbenjamin) [inline sympy/algebras/quaternion.py]: There are too many `=` signs above

File: sympy/algebras/quaternion.py [hunk 2]
```diff
@@ -108,8 +122,18 @@ def from_axis_angle(cls, vector, angle):
 
     @classmethod
     def from_rotation_matrix(cls, M):
-        """Returns the equivalent quaternion of a matrix. The quaternion will be normalized
-        only if the matrix is special orthogonal (orthogonal and det(M) = 1).
+        """Returns the equivalent quaternion of a matrix.
+
+        Parameters
+        ==========
+        M : Matrix
```
Comment (oscarbenjamin) [inline sympy/algebras/quaternion.py]: Should be a blank line between `=====` and `M`.

**Review by oscarbenjamin — COMMENTED** [review]
There are missing blank lines and the `===` don't line up consistently in many places.

File: sympy/algebras/quaternion.py [hunk 3]
```diff
@@ -108,8 +122,18 @@ def from_axis_angle(cls, vector, angle):
...
+            The matrix from which an equivalent quaternion has to be obtained.
+
+        Returns
+        ==========
+        Quaternion
+            The quaternion equivalent to given matrix. The quaternion will be normalized
+            only if the matrix is special orthogonal (orthogonal and det(M) = 1).
 
         Example
```
Comment (oscargus) [inline sympy/algebras/quaternion.py]: Examples (add an s). (Also elsewhere.)

**Comment by oscargus** [comment]
Thanks!

**Pull request merged**